public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx

```diff
@@ -134,33 +147,46 @@ export const AlertRuleForm = ({ existing, prefill }: Props) => {
       }
     }
 
-    dispatch(
-      saveRuleFormAction({
-        values: {
-          ...defaultValues,
-          ...values,
-          annotations:
-            values.annotations
-              ?.map(({ key, value }) => ({ key: key.trim(), value: value.trim() }))
-              .filter(({ key, value }) => !!key && !!value) ?? [],
-          labels:
-            values.labels
-              ?.map(({ key, value }) => ({ key: key.trim(), value: value.trim() }))
-              .filter(({ key }) => !!key) ?? [],
-        },
-        existing,
-        redirectOnSave: exitOnSave ? returnTo : undefined,
-        initialAlertRuleName: defaultValues.name,
-        evaluateEvery: evaluateEvery,
-      })
-    );
+    const ruleDefinition = grafanaTypeRule ? formValuesToRulerGrafanaRuleDTO(values) : formValuesToRulerRuleDTO(values);
+
+    const ruleGroupIdentifier = existing
+      ? getRuleGroupLocationFromRuleWithLocation(existing)
+      : getRuleGroupLocationFromFormValues(values);
+
+    // @TODO what is "evaluateEvery" being used for?
+    // @TODO move this to a hook too to make sure the logic here is tested for regressions?
```

Comment:
Another discussion topic; should this be moved to a "updateOrMove" hook? Maybe we just need the "update" hook but with support for moving a rule? 

Thoughts?

Reply:
I'm okay with the `update` hook with support for moving